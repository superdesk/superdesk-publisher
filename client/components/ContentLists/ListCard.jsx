import React from "react";
import PropTypes from "prop-types";
import classNames from "classnames";
import moment from "moment";
import helpers from "../../services/helpers.js";
import _ from "lodash";

import Dropdown from "../UI/Dropdown";
import Modal from "../UI/Modal";

class ListCard extends React.Component {
  constructor(props) {
    super(props);

    this._isMounted = false;

    this.state = {
      list: { ...props.list },
      isEditing: this.props.list.name ? false : true,
      loading: false,
      moreItemsAmount:
        props.list.content_list_items_count - 5 > 0
          ? props.list.content_list_items_count - 5
          : 0,
      modalType: null
    };
  }

  modalClose = () => {
    this.setState({ modalType: null });
  };

  settingsShow = () => {
    this.setState({ modalType: "settings" });
  };

  save = () => {
    let list = this.state.list.id
      ? helpers.getUpdatedValues(this.state.list, this.props.list)
      : { ...this.state.list };

    this.props.publisher
      .manageList(list, this.state.list.id)
      .then(() => {
        this.modalClose();
        this.editClose();
      })
      .catch(err => {
        console.error(err);
      });
  };

  deleteConfirm = () => {
    this.setState({ modalType: "delete" });
  };

  deleteList = () => {
    this.props.publisher
      .removeList(this.state.list.id)
      .then(() => this.props.onListDelete(this.state.list.id));

    this.modalClose();
  };

  handleInputChange = (e, a) => {
    let { name, value } = e.target;
    let list = { ...this.state.list };

    list[name] = value;
    this.setState({ list });
  };

  edit = () => {
    this.setState({ isEditing: true });
  };

  editClose = () => {
    this.setState({ isEditing: false });
  };

  cancelEditing = () => {
    this.setState({ list: this.props.list, isEditing: false });
  };

  render() {
    let { list, modalType } = this.state;
    let modalContent = "";

    if (modalType === "settings") {
      modalContent = (
        <React.Fragment>
          <div className="modal__header">
            <a className="close" onClick={this.modalClose}>
              <i className="icon-close-small" />
            </a>
            <h3>Settings</h3>
          </div>
          <div className="modal__body">
            <form name="settingsForm">
              <fieldset>
                <div className="field">
                  <label htmlFor="listLimit">number of articles limit</label>
                  <input
                    type="number"
                    className="line-input"
                    value={this.state.list.limit}
                    name="limit"
                    min="0"
                    onChange={this.handleInputChange}
                  />
                </div>
                <div className="field">
                  <label htmlFor="listCacheLifeTime">cache lifetime</label>
                  <input
                    type="number"
                    className="line-input"
                    name="cache_life_time"
                    value={this.state.list.cache_life_time}
                    onChange={this.handleInputChange}
                    min="0"
                    required=""
                  />
                </div>
              </fieldset>
            </form>
          </div>
          <div className="modal__footer">
            <button className="btn" onClick={this.modalClose}>
              Cancel
            </button>
            <button
              className="btn btn--primary"
              onClick={this.save}
              disabled={_.isEmpty(
                helpers.getUpdatedValues(this.state.list, this.props.list)
              )}
            >
              Save
            </button>
          </div>
        </React.Fragment>
      );
    }

    if (modalType === "delete") {
      modalContent = (
        <React.Fragment>
          <div className="modal__header ">
            <h3>Confirm</h3>
          </div>
          <div className="modal__body ">
            Please confirm you want to delete list.
          </div>
          <div className="modal__footer">
            <button className="btn" onClick={this.modalClose}>
              Cancel
            </button>
            <button className="btn btn--primary" onClick={this.deleteList}>
              OK
            </button>
          </div>
        </React.Fragment>
      );
    }

    return (
      <React.Fragment>
        <div className="sd-card flexGrow">
          <div
            className={classNames("sd-card__header sd-card__header--padding", {
              "sd-card__header--secondary-color": list.type === "automatic"
            })}
          >
            {this.state.isEditing ? (
              <React.Fragment>
                <div className="sd-card__heading">
                  <input
                    type="text"
                    name="name"
                    onChange={this.handleInputChange}
                    value={this.state.list.name}
                    className="line-input line-input--alt"
                    required
                  />
                </div>
                <div className="sd-card__btn-group sd-card__btn-group--right margin--top0">
                  <button
                    onClick={this.cancelEditing}
                    className="btn btn--icon-only"
                  >
                    <i className="icon-close-small" />
                  </button>
                  <button onClick={this.save} className="btn btn--icon-only">
                    <i className="icon-ok" />
                  </button>
                </div>
              </React.Fragment>
            ) : (
              <React.Fragment>
                <div className="sd-card__heading" onClick={this.edit}>
                  {list.name}
                </div>
                <div className="sd-card__actions-group">
                  <a
                    className="btn btn--small btn--hollow btn--ui-dark"
                    onClick={() => this.props.listEdit(list)}
                  >
                    Edit
                  </a>
                  <Dropdown
                    button={
                      <button className="dropdown__toggle icn-btn">
                        <i className="icon-dots-vertical" />
                      </button>
                    }
                  >
                    <li>
                      <button onClick={this.settingsShow} title="Settings">
                        <i className="icon-settings" />
                        Settings
                      </button>
                    </li>
                    <li>
                      <button onClick={this.deleteConfirm} title="Remove list">
                        <i className="icon-trash" />
                        Remove
                      </button>
                    </li>
                  </Dropdown>
                </div>
              </React.Fragment>
            )}
          </div>
          <div className="sd-card__content sd-card__content--scrollable relative">
            <ul className="sd-card__content-list">
              {this.state.loading && <div className="sd-loader" />}
              {!this.state.loading &&
                (!this.state.list.latest_items ||
                  !this.state.list.latest_items.length) && (
                  <div className="sd-card__content-list-item sd-card__content-list-item--small">
                    <span>No articles in this list</span>
                  </div>
                )}
              {!this.state.loading &&
              this.state.list.latest_items &&
              this.state.list.latest_items.length
                ? this.state.list.latest_items.map((article, index) => (
                    <li
                      key={
                        "listElement" +
                        article.content.id +
                        this.state.list.id +
                        index
                      }
                      className="sd-card__content-list-item sd-card__content-list-item--small"
                    >
                      {article.content.title}
                    </li>
                  ))
                : null}
              {!this.state.loading &&
              this.state.list.latest_items &&
              this.state.list.latest_items.length &&
              this.state.moreItemsAmount ? (
                <li
                  className="sd-card__content-list-item sd-card__content-list-item--small"
                  style={{ fontStyle: "italic" }}
                >
                  {this.state.moreItemsAmount} more...
                </li>
              ) : null}
            </ul>
          </div>
          <div className="sd-card__footer sd-card__footer--spread">
            <div>
              <span className="sd-text__info">Last modified: </span>
              <span>{moment(list.updated_at).fromNow()}</span>
            </div>
            {list.enabled && (
              <span className="label label--success label--hollow">active</span>
            )}
          </div>
        </div>

        <Modal isOpen={this.state.modalType ? true : false}>
          {modalContent}
        </Modal>
      </React.Fragment>
    );
  }
}

ListCard.propTypes = {
  list: PropTypes.object.isRequired,
  publisher: PropTypes.object.isRequired,
  listEdit: PropTypes.func.isRequired
};

export default ListCard;
