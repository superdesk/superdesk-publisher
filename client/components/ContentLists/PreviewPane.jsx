import React from "react";
import PropTypes from "prop-types";
import _ from "lodash";

import ArticlePreview from "../generic/ArticlePreview";

class PreviewPane extends React.Component {
  constructor(props) {
    super(props);

    this._isMounted = false;

    this.state = {
      article: null
    };
  }

  componentDidMount() {
    this._isMounted = true;
  }

  componentWillUnmount() {
    this._isMounted = false;
  }

  componentDidUpdate(prevProps) {
    if (
      !_.isEqual(prevProps.article, this.props.article) &&
      this.props.article
    ) {
      this.loadArticle();
    }
  }

  loadArticle = () => {
    this.props.publisher.setTenant(this.props.article.tenant);

    this.props.publisher.getArticle(this.props.article.id).then(response => {
      if (this._isMounted) {
        this.setState({
          article: response
        });
      }
    });
  };

  render() {
    if (!this.props.article) return null;
    let article = this.props.article;
    article.slideshows = null;

    if (this.state.article) article = this.state.article;

    return <ArticlePreview article={article} close={this.props.close} />;
  }
}

PreviewPane.propTypes = {
  close: PropTypes.func.isRequired,
  article: PropTypes.object,
  publisher: PropTypes.object.isRequired
};

export default PreviewPane;
