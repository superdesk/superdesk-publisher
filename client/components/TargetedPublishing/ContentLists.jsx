import React from "react";
import PropTypes from "prop-types";
import { Button } from "superdesk-ui-framework/react";
import { gettext } from "../../superdeskApi";
import ContentListElement from "./ContentListElement";

const ContentLists = (props) => {
  let ruleLists = [...props.ruleLists];
  let allContentLists = [...props.contentLists];
  let remainingLists = [...props.contentLists];

  ruleLists.forEach((list) => {
    let index = remainingLists.findIndex((rlist) => list.id === rlist.id);
    if (index >= 0) {
      remainingLists.splice(index, 1);
    }
  });

  return (
    <div style={{ margin: "1em 0" }}>
      <label className="form-label">{gettext("Content lists")}</label>
      {ruleLists.map((list, index) => (
        <ContentListElement
          key={"list_el" + index}
          ruleLists={props.ruleLists}
          removeList={props.removeList}
          index={index}
          list={list}
          allContentLists={allContentLists}
          save={props.save}
          ruleLists={props.ruleLists}
        />
      ))}

      {!!remainingLists.length && (
        <Button
          onClick={props.addList}
          text={gettext("Add to content list")}
          style="hollow"
          type="primary"
          expand="true"
        />
      )}
    </div>
  );
};

ContentLists.propTypes = {
  ruleLists: PropTypes.array.isRequired,
  contentLists: PropTypes.array.isRequired,
  removeList: PropTypes.func.isRequired,
  addList: PropTypes.func.isRequired,
  save: PropTypes.func.isRequired,
};

export default ContentLists;
