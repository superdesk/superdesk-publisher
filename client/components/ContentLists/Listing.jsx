import React from "react";
import PropTypes from "prop-types";

import CheckButton from "../UI/CheckButton";
import ListCard from "./ListCard";
import Dropdown from "../UI/Dropdown";

import SearchBar from "../UI/SearchBar";

class Listing extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      filter: "all",
      search: ""
    };
  }

  setFilter = kind => this.setState({ filter: kind });

  addList = type => {
    if (this.state.filter !== "all" && type !== this.state.filter) {
      this.setState({ filter: "all" });
    }
    this.props.addList(type);
  };

  render() {
    let lists = [...this.props.lists];

    if (this.state.filter !== "all")
      lists = lists.filter(list => list.type === this.state.filter);

    if (this.state.search)
      lists = lists.filter(list =>
        list.name.toLowerCase().includes(this.state.search)
      );

    return (
      <div className="sd-column-box__main-column relative sd-display-flex-column">
        <div className="subnav subnav--lower-z-index">
          <SearchBar
            value={this.state.search}
            onChange={value => this.setState({ search: value.toLowerCase() })}
            debounceTime={1}
          />
          <CheckButton
            label="All"
            isChecked={this.state.filter === "all"}
            onClick={() => this.setFilter("all")}
          />
          <CheckButton
            label="Automatic"
            isChecked={this.state.filter === "automatic"}
            onClick={() => this.setFilter("automatic")}
          />
          <CheckButton
            label="Manual"
            isChecked={this.state.filter === "manual"}
            onClick={() => this.setFilter("manual")}
          />

          <div className="subnav__stretch-bar" />
          <Dropdown
            button={
              <button
                className="navbtn dropdown sd-create-btn dropdown-toggle"
                sd-tooltip="Create new list"
              >
                <i className="icon-plus-large" />
                <span className="circle" />
              </button>
            }
          >
            <li>
              <div className="dropdown__menu-label">Create new list</div>
            </li>
            <li className="dropdown__menu-divider" />
            <li>
              <button onClick={() => this.addList("automatic")}>
                Automatic List
              </button>
            </li>
            <li>
              <button onClick={() => this.addList("manual")}>
                Manual List
              </button>
            </li>
          </Dropdown>
        </div>
        <div className="sd-display-flex-column">
          <div className="sd-grid-list sd-grid-list--large">
            {lists.map(list => (
              <ListCard
                key={list.id + list.name + "sf"}
                list={list}
                publisher={this.props.publisher}
                onListDelete={id => this.props.onListDelete(id)}
                listEdit={list => this.props.listEdit(list)}
              />
            ))}
          </div>
        </div>
        {this.props.lists && !this.props.lists.length ? (
          <div className="panel-info">
            <div className="panel-info__icon">
              <i className="big-icon--add-to-list" />
            </div>
            <h3 className="panel-info__heading">No Content Lists</h3>
            <p className="panel-info__description">Create the first one...</p>
          </div>
        ) : null}
      </div>
    );
  }
}

Listing.propTypes = {
  lists: PropTypes.array.isRequired,
  publisher: PropTypes.object.isRequired,
  onListDelete: PropTypes.func.isRequired,
  addList: PropTypes.func.isRequired,
  listEdit: PropTypes.func.isRequired
};

export default Listing;
