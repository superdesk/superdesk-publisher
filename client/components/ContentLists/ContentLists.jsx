import React from "react";
import PropTypes from "prop-types";
import classNames from "classnames";
import _ from "lodash";

import SitesSideNav from "../generic/SitesSideNav";
import Listing from "./Listing";
import AutomaticList from "./Automatic/Automatic";
import ManualList from "./Manual/Manual";
import PreviewPane from "./PreviewPane";

class ContentLists extends React.Component {
  constructor(props) {
    super(props);

    this._isMounted = false;

    this.state = {
      loading: true,
      sites: [],
      selectedSite: {},
      lists: [],
      tenantsNavOpen: true,
      selectedList: null,
      filtersOpen: false,
      previewOpen: false,
      previewItem: null
    };
  }

  componentDidMount() {
    this._isMounted = true;

    this.props.publisher
      .setToken()
      .then(this.props.publisher.querySites)
      .then(sites => {
        let tenantsNavOpen = sites.length > 1 ? true : false;
        this.setState({ sites, tenantsNavOpen });

        if (this.props.tenant) {
          let site = sites.find(site => site.code === this.props.tenant);
          if (site) this.setTenant(site);
        } else if (sites[0]) {
          this.setTenant(sites[0]);
        }
      });
  }

  componentWillUnmount() {
    this._isMounted = false;
  }

  setTenant = site => {
    this.props.publisher.setTenant(site);

    this.setState(
      {
        selectedSite: site
      },
      this._getLists
    );
  };

  listEdit = list =>
    this.setState({
      selectedList: list,
      filtersOpen: list.type === "automatic" ? true : false
    });

  cancelListEdit = () =>
    this.setState({
      selectedList: null,
      filtersOpen: false,
      previewOpen: false
    });

  _getLists = () => {
    this.setState({ loading: true }, () => {
      return this.props.publisher.queryLists().then(lists => {
        let selectedList = null;

        if (this.props.list) {
          let list = lists.find(l => l.id === parseInt(this.props.list));
          if (list) selectedList = list;
        }

        lists = _.orderBy(lists, "name", "asc");

        if (this._isMounted)
          this.setState({
            lists,
            loading: false,
            selectedList,
            filtersOpen:
              selectedList && selectedList.type === "automatic" ? true : false
          });
      });
    });
  };

  onListDelete = id => {
    let lists = [...this.state.lists];
    let index = lists.findIndex(list => list.id === id);

    if (index > -1) {
      delete lists[index];

      this.setState({ lists });
    }
  };

  onListUpdate = updatedList => {
    let lists = [...this.state.lists];
    let index = lists.findIndex(list => list.id === updatedList.id);
    let selectedList = { ...this.state.selectedList };

    if (selectedList.id === updatedList.id) {
      selectedList = updatedList;
    }

    if (index > -1) {
      lists[index] = updatedList;
    }
    this.setState({ lists, selectedList });
  };

  toggleTenantsNav = () =>
    this.setState({ tenantsNavOpen: !this.state.tenantsNavOpen });

  toggleFilters = () => this.setState({ filtersOpen: !this.state.filtersOpen });

  openPreview = item => this.setState({ previewOpen: true, previewItem: item });
  closePreview = () => this.setState({ previewOpen: false, previewItem: null });

  addList = type => {
    let list = {
      name: "",
      type: type,
      cache_life_time: 0
    };

    this.setState({ lists: [...this.state.lists, list] });
  };

  render() {
    return (
      <React.Fragment>
        {this.state.loading && <div className="sd-loader" />}
        <div
          className={classNames("sd-page-content__content-block", {
            "sd-page-content__content-block--double-sidebar": this.state
              .selectedList,
            "open-filters": this.state.filtersOpen,
            "open-preview": this.state.previewOpen
          })}
        >
          <div className="subnav">
            <a
              href="#/publisher"
              className="navbtn navbtn--left"
              sd-tooltip="Dashboard"
              flow="right"
            >
              <i className="icon-arrow-left" />
            </a>
            <h3 className="subnav__page-title">
              Content Lists
              {this.state.selectedSite && (
                <span> / {this.state.selectedSite.name}</span>
              )}
            </h3>
          </div>
          <div
            className={classNames("sd-column-box--3", {
              "content-nav-closed": !this.state.tenantsNavOpen
            })}
          >
            <SitesSideNav
              sites={this.state.sites}
              selectedSite={this.state.selectedSite}
              setTenant={site => this.setTenant(site)}
              toggle={this.toggleTenantsNav}
              open={this.state.tenantsNavOpen}
            />

            {!this.state.selectedList && (
              <Listing
                lists={this.state.lists}
                publisher={this.props.publisher}
                onListDelete={id => this.onListDelete(id)}
                addList={type => this.addList(type)}
                listEdit={list => this.listEdit(list)}
              />
            )}

            {this.state.selectedList &&
              this.state.selectedList.type === "automatic" && (
                <AutomaticList
                  list={this.state.selectedList}
                  lists={this.state.lists}
                  publisher={this.props.publisher}
                  listEdit={list => this.listEdit(list)}
                  onEditCancel={this.cancelListEdit}
                  onListUpdate={list => this.onListUpdate(list)}
                  toggleFilters={this.toggleFilters}
                  openPreview={item => this.openPreview(item)}
                  previewItem={this.state.previewItem}
                  filtersOpen={this.state.filtersOpen}
                  api={this.props.api}
                />
              )}

            {this.state.selectedList &&
              this.state.selectedList.type === "manual" && (
                <ManualList
                  list={this.state.selectedList}
                  lists={this.state.lists}
                  publisher={this.props.publisher}
                  listEdit={list => this.listEdit(list)}
                  onEditCancel={this.cancelListEdit}
                  onListUpdate={list => this.onListUpdate(list)}
                  toggleFilters={this.toggleFilters}
                  openPreview={item => this.openPreview(item)}
                  previewItem={this.state.previewItem}
                  filtersOpen={this.state.filtersOpen}
                  api={this.props.api}
                />
              )}

            <PreviewPane
              article={
                this.state.previewItem && this.state.previewItem.content
                  ? this.state.previewItem.content
                  : this.state.previewItem
              }
              close={this.closePreview}
              publisher={this.props.publisher}
            />
          </div>
        </div>
      </React.Fragment>
    );
  }
}

ContentLists.propTypes = {
  tenant: PropTypes.string,
  list: PropTypes.string,
  publisher: PropTypes.object.isRequired,
  api: PropTypes.func.isRequired
};

export default ContentLists;
