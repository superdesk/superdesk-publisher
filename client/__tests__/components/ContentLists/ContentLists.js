import React from "react";
import ContentLists from "../../../components/ContentLists/ContentLists";
import { render, waitForElement } from "@testing-library/react";
import axios from "axios";

jest.mock("react-virtualized/styles.css", () => {
  return {};
});

import Publisher from "../../../__mocks__/publisher";

const publisher = new Publisher();

jest.mock("axios");

let api = () => {};

describe("ContentLists/ContentLists", () => {
  it("renders properly", async () => {
    const { container, getByText } = render(
      <ContentLists tenant={null} list={null} api={api} publisher={publisher} />
    );
    let title = await waitForElement(() => getByText("Content Lists"));
    expect(container.firstChild).toMatchSnapshot();
  });
});
