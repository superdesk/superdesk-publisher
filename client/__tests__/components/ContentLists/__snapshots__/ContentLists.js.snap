// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentLists/ContentLists renders properly 1`] = `
<div
  class="sd-page-content__content-block"
>
  <div
    class="subnav"
  >
    <a
      class="navbtn navbtn--left"
      flow="right"
      href="#/publisher"
      sd-tooltip="Dashboard"
    >
      <i
        class="icon-arrow-left"
      />
    </a>
    <h3
      class="subnav__page-title"
    >
      Content Lists
      <span>
         / 
        Site1
      </span>
    </h3>
  </div>
  <div
    class="sd-column-box--3"
  >
    <div
      class="nav-tabs-vertical nav-tabs-vertical--closed"
    >
      <ul
        class="nav-tabs-vertical__list"
      >
        <li
          class="nav-tabs-vertical__tab"
        >
          <button
            class="nav-tabs-vertical__link"
          >
            Website
          </button>
        </li>
      </ul>
    </div>
    <div
      class="sd-content-navigation-panel sd-content-navigation-panel--border-right"
    >
      <div
        class="subnav subnav--padded subnav--lower-z-index"
      >
        <h3
          class="sd-content-nav-title sd-content-nav-title--uppercase"
        >
          Website
        </h3>
        <a
          class="icn-btn ml-auto"
        >
          <i
            class="icon-close-small"
          />
        </a>
      </div>
      <nav
        class="sd-content-nav"
      >
        <ul>
          <li>
            <a
              class="sd-content-nav__btn sd-content-nav__btn--active"
              style="cursor: pointer;"
            >
              Site1
            </a>
          </li>
          <li>
            <a
              class="sd-content-nav__btn"
              style="cursor: pointer;"
            >
              Site2
            </a>
          </li>
        </ul>
      </nav>
    </div>
    <div
      class="sd-column-box__main-column relative sd-display-flex-column"
    >
      <div
        class="subnav subnav--lower-z-index"
      >
        <div
          class="flat-searchbar"
        >
          <div
            class="search-handler"
          >
            <label
              class="trigger-icon"
              for="search-input"
            >
              <i
                class="icon-search"
              />
            </label>
            <input
              placeholder="Search"
              type="text"
              value=""
            />
          </div>
        </div>
        <span
          class="sd-check__wrapper"
        >
          <span
            class="sd-checkbox sd-checkbox--button-style sd-checkbox--radio checked"
          >
            <label>
              All
            </label>
          </span>
        </span>
        <span
          class="sd-check__wrapper"
        >
          <span
            class="sd-checkbox sd-checkbox--button-style sd-checkbox--radio"
          >
            <label>
              Automatic
            </label>
          </span>
        </span>
        <span
          class="sd-check__wrapper"
        >
          <span
            class="sd-checkbox sd-checkbox--button-style sd-checkbox--radio"
          >
            <label>
              Manual
            </label>
          </span>
        </span>
        <div
          class="subnav__stretch-bar"
        />
        <div
          class="dropdown dropdown--align-right"
          flow="left"
        >
          <span>
            <button
              class="navbtn dropdown sd-create-btn dropdown-toggle"
              sd-tooltip="Create new list"
            >
              <i
                class="icon-plus-large"
              />
              <span
                class="circle"
              />
            </button>
          </span>
          <ul
            class="dropdown__menu dropdown--align-right"
          >
            <li>
              <div
                class="dropdown__menu-label"
              >
                Create new list
              </div>
            </li>
            <li
              class="dropdown__menu-divider"
            />
            <li>
              <button>
                Automatic List
              </button>
            </li>
            <li>
              <button>
                Manual List
              </button>
            </li>
          </ul>
        </div>
      </div>
      <div
        class="sd-display-flex-column"
      >
        <div
          class="sd-grid-list sd-grid-list--large"
        >
          <div
            class="sd-card flexGrow"
          >
            <div
              class="sd-card__header sd-card__header--padding sd-card__header--secondary-color"
            >
              <div
                class="sd-card__heading"
              >
                Most shared
              </div>
              <div
                class="sd-card__actions-group"
              >
                <a
                  class="btn btn--small btn--hollow btn--ui-dark"
                >
                  Edit
                </a>
                <div
                  class="dropdown dropdown--align-right"
                  flow="left"
                >
                  <span>
                    <button
                      class="dropdown__toggle icn-btn"
                    >
                      <i
                        class="icon-dots-vertical"
                      />
                    </button>
                  </span>
                  <ul
                    class="dropdown__menu dropdown--align-right"
                  >
                    <li>
                      <button
                        title="Settings"
                      >
                        <i
                          class="icon-settings"
                        />
                        Settings
                      </button>
                    </li>
                    <li>
                      <button
                        title="Remove list"
                      >
                        <i
                          class="icon-trash"
                        />
                        Remove
                      </button>
                    </li>
                  </ul>
                </div>
              </div>
            </div>
            <div
              class="sd-card__content sd-card__content--scrollable relative"
            >
              <ul
                class="sd-card__content-list"
              >
                <div
                  class="sd-card__content-list-item sd-card__content-list-item--small"
                >
                  <span>
                    No articles in this list
                  </span>
                </div>
              </ul>
            </div>
            <div
              class="sd-card__footer sd-card__footer--spread"
            >
              <div>
                <span
                  class="sd-text__info"
                >
                  Last modified: 
                </span>
                <span>
                  2 days ago
                </span>
              </div>
              <span
                class="label label--success label--hollow"
              >
                active
              </span>
            </div>
          </div>
          <div
            class="modal modal--double"
            data-backdrop="static"
          >
            <div
              class="modal__dialog"
            >
              <div
                class="modal__content"
              />
            </div>
          </div>
          <div
            class="sd-card flexGrow"
          >
            <div
              class="sd-card__header sd-card__header--padding"
            >
              <div
                class="sd-card__heading"
              >
                Top news
              </div>
              <div
                class="sd-card__actions-group"
              >
                <a
                  class="btn btn--small btn--hollow btn--ui-dark"
                >
                  Edit
                </a>
                <div
                  class="dropdown dropdown--align-right"
                  flow="left"
                >
                  <span>
                    <button
                      class="dropdown__toggle icn-btn"
                    >
                      <i
                        class="icon-dots-vertical"
                      />
                    </button>
                  </span>
                  <ul
                    class="dropdown__menu dropdown--align-right"
                  >
                    <li>
                      <button
                        title="Settings"
                      >
                        <i
                          class="icon-settings"
                        />
                        Settings
                      </button>
                    </li>
                    <li>
                      <button
                        title="Remove list"
                      >
                        <i
                          class="icon-trash"
                        />
                        Remove
                      </button>
                    </li>
                  </ul>
                </div>
              </div>
            </div>
            <div
              class="sd-card__content sd-card__content--scrollable relative"
            >
              <ul
                class="sd-card__content-list"
              >
                <li
                  class="sd-card__content-list-item sd-card__content-list-item--small"
                >
                  seo test
                </li>
              </ul>
            </div>
            <div
              class="sd-card__footer sd-card__footer--spread"
            >
              <div>
                <span
                  class="sd-text__info"
                >
                  Last modified: 
                </span>
                <span>
                  2 days ago
                </span>
              </div>
              <span
                class="label label--success label--hollow"
              >
                active
              </span>
            </div>
          </div>
          <div
            class="modal modal--double"
            data-backdrop="static"
          >
            <div
              class="modal__dialog"
            >
              <div
                class="modal__content"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
